ir/instr.cpp

```diff
@@ -482,6 +486,28 @@ StateValue BinOp::toSMT(State &s) const {
                        false);
     };
     break;
+
+  case FMinNum:
+  case FMaxNum:
+    fn = [&](auto a, auto ap, auto b, auto bp) -> StateValue {
+      expr ndet = expr::mkFreshVar("maxminnondet", true);
+      s.addQuantVar(ndet);
+      auto ndz = expr::mkIf(ndet, expr::mkNumber("0", a),
+                            expr::mkNumber("-0", a));
+      
+      auto v = [&](expr &a, expr &b) {
+        expr z = a.isFPZero() && b.isFPZero();
+        // (+, -) or (-, +)
+        expr s = (a.isFPNeg() && !b.isFPNeg()) || (!a.isFPNeg() && b.isFPNeg());
+        expr cmp = (op == FMinNum) ? a.fole(b) : a.foge(b);
+        return expr::mkIf(a.isNaN(), b,
+                          expr::mkIf(b.isNaN(), a,
+                                     expr::mkIf(z && s, ndz,
```

Comment:
from my reading of langref, fmax(-0, -0) can return +0. If you agree, then `s` isn't needed here

Reply:
let me run TV on transforms suite again without s, and let's see if it is breaking anything.

Reply:
just to avoid spurious errors.

Reply:
yeah, allowing non-det +0, -0 for input (+0, +0) and (-0, -0) eliminates three Unexpected Failures when running TV on Transform suite.